### PR TITLE
Deploy Sara to K8s

### PR DESCRIPTION
**Pull request by melindaloubser1**
Add workflow to deploy Sara to K8s automatically

### CODE REVIEW

**Comment by melindaloubser1** [comment]
I switched to this file change filter https://github.com/RasaHQ/pr-changed-files-filter that we use in rasa as well, but now it seems to be ignoring my conditions 😬

**Comment by melindaloubser1** [comment]
Because of the changes to CI/CD, the unit tests workflow should be merged first.

**Comment by erohmensing** [comment]
@melindaloubser1 does this need a review?

**Comment by melindaloubser1** [comment]
Yes! didn't realize I draft marked it

File: .github/deployments/helmfile.yaml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+          versions:
+            rasaProduction:
+              resources:
+                requests:
+                  memory: {{ .Values.rasaProduction.resources.requests.memory }}
+                  cpu: {{ .Values.rasaProduction.resources.requests.cpu}}
+            rasaWorker:
+              nodeSelector:
+                nodeFeature: extraMemory
+              resources:
```
Comment (melindaloubser1) [inline .github/deployments/helmfile.yaml:98]: @desmarchris

**Review by erohmensing — APPROVED** [review]
an absolute bear of a PR, awesome 🚀

File: .github/workflows/continuous_deployment.yml [hunk 2]
```diff
@@ -2,15 +2,57 @@ name: Continuous Deployment
 on:
   push: 
     branches:
-    - 'main'
+    - 'k8s' #TODO: main
```
Comment (melindaloubser1) [inline .github/workflows/continuous_deployment.yml:5]:
```suggestion
    -  'main'
```

**Pull request merged**